handler.go

```diff
@@ -64,6 +72,7 @@ type Handler struct {
 	Next   http.Handler
 	Logger Logger
 	Skip   func(r *http.Request) bool
+	Extra  func(w http.ResponseWriter, r *http.Request) map[string]string
```

Comment:
I think we don't need `http.ResponseWriter` as that could let someone write the response